src/proxy.rs

```diff
@@ -24,21 +29,20 @@ impl Proxy {
         server_transport: T,
         client_transport: U,
     ) -> Box<dyn Future<Item = (), Error = io::Error> + Send> {
-        let jet_sink_server = server_transport.message_sink();
-        let mut jet_stream_server = server_transport.message_stream();
+        let (client_writer, server_reader) = bip_buffer_with_len(BIP_BUFFER_LEN);
```

Comment:
Create bip buffer's reader and writer for the server and client, and pass them to the split_transport methods.